config/config.go

```diff
@@ -1,16 +1,139 @@
+// This source code file is AUTO-GENERATED by github.com/taskcluster/jsonschema2go
+
 package config
 
-type Config struct {
-	Taskcluster struct {
-		ClientId    string
-		AccessToken string
-		Certificate string
+import "github.com/taskcluster/taskcluster-worker/runtime"
+
+type (
+	// The global configuration settings used by taskcluster-worker, not particular
+	// to any engine or plugin.
+	Config struct {
+
+		// The number of tasks that this worker supports running in parallel.
+		Capacity int `json:"capacity"`
+
+		// The set of credentials that should be used by the worker when
+		// authenticating against taskcluster endpoints.
+		Credentials struct {
+
+			// The security-sensitive access token for the client. Do not share or expose!
+			//
+			// Syntax:     ^[a-zA-Z0-9_-]{22,66}$
+			AccessToken string `json:"accessToken"`
+
+			// The certificate for the client, if using temporary credentials and/or using authorized scopes.
+			Certificate string `json:"certificate"`
+
+			// The Client ID for the client. Not very helpful, am I?
+			//
+			// Syntax:     ^[A-Za-z0-9@/:._-]+$
+			ClientId string `json:"clientId"`
+		} `json:"credentials"`
+
+		// The provisioner (if any) that is responsible for spawning instances of this worker. Typically `aws-provisioner-v1`.
+		ProvisionerId string `json:"provisionerId"`
+
+		// Configuration relating to the polling of the TaskCluster Queue.
+		QueueService struct {
+
+			// The number of seconds prior to the task claim expiry that tasks should be reclaimed.
+			ExpirationOffset int `json:"expirationOffset"`
+		} `json:"queueService"`
+
+		// A logical group for this worker to belong to, such as an AWS region.
+		WorkerGroup string `json:"workerGroup"`
+
+		// A unique name that can be used to identify which worker instance this is (such as AWS instance id).
+		WorkerId string `json:"workerId"`
 	}
-	ProvisionerId string
-	WorkerGroup   string
-	WorkerId      string
-	Capacity      int
-	QueueService  struct {
-		ExpirationOffset int
+)
+
+var ConfigSchema = func() runtime.CompositeSchema {
+	schema, err := runtime.NewCompositeSchema(
+		"config",
+		`
+		{
+		  "$schema": "http://json-schema.org/draft-04/schema#",
+		  "additionalProperties": false,
+		  "description": "The global configuration settings used by taskcluster-worker, not particular\nto any engine or plugin.",
+		  "properties": {
+		    "capacity": {
+		      "description": "The number of tasks that this worker supports running in parallel.",
+		      "title": "Capacity",
+		      "type": "integer"
+		    },
+		    "credentials": {
```

Comment:
we can tweak this later, but I wonder if there will be other credentails we will be storing on the worker so it might make sense to put it under it's own key